### PR TITLE
Py3 ddsim

### PR DESCRIPTION
**Pull request by andresailer**
BEGINRELEASENOTES
- ddsim: fix problem with `--dumpSteeringFile` in python3
- ddsim: fix issue in where isotrop was False, when distribution was set (python3)
- ddsim: fix exception in exceptions for gun direction and position
- ddsim: fix problem where `hepmc.useHepMC3` was always True by default
- PluginManager: rename the listcomponents executable to listcomponents_dd4hep, to avoid clash with the Gaudi listcomponents. This is transparent for users of the dd4hep_add_plugin cmake macro.
- DDG4Dict: remove `dd4hep::sim::Geant4InputAction::Particles;` as this is available under a different name, fixes warning
- CMake: DD4hepConfig.cmake: use find_dependency with CONFIG for Geant4, TBB, ROOT, LCIO, and if needed CLHEP, to avoid accidental use of `Find<PACKAGE>.cmake` modules

ENDRELEASENOTES

### CODE REVIEW

File: DDG4/python/DDSim/DD4hepSimulation.py [hunk 1]
```diff
@@ -733,20 +733,46 @@ def _buildInputStage(self, simple, generator_input_modules, output_level=None, h
...
-  if isinstance(parTypeA, ConfigHelper) and isinstance(parTypeB, ConfigHelper):
-    return 1 if str(parA[0]) > str(parB[0]) else -1
-  elif isinstance(parTypeA, ConfigHelper):
-    return 1
-  elif isinstance(parTypeB, ConfigHelper):
-    return -1
+
+def sortParameters(key):
+  if six.PY3:
+    from functools import cmp_to_key
```
Comment (petricm) [inline DDG4/python/DDSim/DD4hepSimulation.py:739]: What's from with the `cmp_to_key` from py2?

Comment (andresailer) [reply DDG4/python/DDSim/DD4hepSimulation.py:739]: ahh, I read this was added in python 3.7 :wallb:
https://docs.python.org/2/howto/sorting.html#the-old-way-using-the-cmp-parameter

**Comment by andresailer** [comment]
rebased

File: cmake/DD4hepConfig.cmake.in [hunk 2]
```diff
@@ -60,18 +60,18 @@ if(DD4HEP_USE_GEANT4)
...
 if(DD4HEP_USE_LCIO)
   MESSAGE(STATUS "@CMAKE_PROJECT_NAME@ uses LCIO")
-  find_dependency(LCIO REQUIRED)
+  find_dependency(LCIO REQUIRED CONFIG)
   DD4HEP_SETUP_LCIO_TARGETS()
 endif()
 
 if(DD4HEP_USE_TBB)
   MESSAGE(STATUS "@CMAKE_PROJECT_NAME@ uses TBB")
-  find_dependency(TBB REQUIRED)
```
Comment (petricm) [inline cmake/DD4hepConfig.cmake.in:74]: what bout adding this also in the main cmakelists?

Comment (andresailer) [reply cmake/DD4hepConfig.cmake.in:74]: Yes, done.

**Review by petricm — APPROVED** [review]
Just add to the release notes where all the CONFIG was added

**Comment by andresailer** [comment]
updated the release notes for the added CONFIGs

**Pull request merged**